pkg/gui/filetree/file_manager_test.go

```diff
@@ -89,3 +89,62 @@ func TestRender(t *testing.T) {
 		})
 	}
 }
+
+func TestFilterAction(t *testing.T) {
+	scenarios := []struct {
+		name     string
+		filter   FileManagerDisplayFilter
+		files    []*models.File
+		expected []*models.File
+	}{
+		{
+			name:   "filter files with unstaged changes",
+			filter: DisplayUnstaged,
+			files: []*models.File{
+				{Name: "dir2/dir2/file4", ShortStatus: "M ", HasUnstagedChanges: true},
+				{Name: "dir2/file5", ShortStatus: "M ", HasStagedChanges: true},
+				{Name: "file1", ShortStatus: "M ", HasUnstagedChanges: true},
+			},
+			expected: []*models.File{
+				{Name: "dir2/dir2/file4", ShortStatus: "M ", HasUnstagedChanges: true},
+				{Name: "file1", ShortStatus: "M ", HasUnstagedChanges: true},
+			},
+		},
+		{
+			name:   "filter files with staged changes",
+			filter: DisplayStaged,
+			files: []*models.File{
+				{Name: "dir2/dir2/file4", ShortStatus: "M ", HasStagedChanges: true},
+				{Name: "dir2/file5", ShortStatus: "M ", HasStagedChanges: false},
+				{Name: "file1", ShortStatus: "M ", HasStagedChanges: true},
+			},
+			expected: []*models.File{
+				{Name: "dir2/dir2/file4", ShortStatus: "M ", HasStagedChanges: true},
+				{Name: "file1", ShortStatus: "M ", HasStagedChanges: true},
+			},
+		},
+		{
+			name:   "filter all files",
```

Comment:
😆 display all files? I'm setting the filter to see all files :D... no idea ^^ naming stuff is hard :D

Reply:
'filter all files' works for me haha